### DISCUSSION TITLE
Frontend receives wrong type

### DISCUSSION THREAD

**Discussion by madsh93** [nuxt/nuxt#25868] [Questions] [answered]
Anyone know why I am not getting the correct types on the front end when fetching from the Nitro backend?

Nitro API:
```js
return {
  selected: {
    size: selectedSize,
    edition: {
      name: selectedEdition.name, <-- name: string
    },
  }
}
````

Frontend:
```js
selected: {
    size: {
        size: "short" | "standard" | "long" | "extra-long";
        description: string;
    };
    edition: {
        [x: string]: never; <-- Why this? 
    };
}
```

If I hardcode the edition.name I get the correct type. Seems to mess up everytime I use "selectedEdition".

**Comment by TheAlexLichter** [comment]
Could you provide a simple stackblitz repro?

**Reply by madsh93** [reply]
I'm unable to recreate the issue. However, the data missing is coming from functions inside the `server/utils`. As soon as I move those function inside the API it works correctly.

**Reply by madsh93** [reply, marked as answer]
Finally figured it out.
I am using layers, and I had the wrong import. I need to use the relative path:

![CleanShot 2024-02-20 at 11 01 23@2x](https://github.com/nuxt/nuxt/assets/8860723/416cdf78-72ec-457f-bfca-19173b082482)
